application/controllers/SearchController.php

```diff
@@ -3,23 +3,21 @@
 
 namespace Icinga\Controllers;
 
-use Icinga\Web\Controller\ActionController;
-use Icinga\Web\Widget;
 use Icinga\Web\Widget\SearchDashboard;
+use ipl\Web\Compat\CompatController;
 
 /**
  * Search controller
  */
-class SearchController extends ActionController
+class SearchController extends CompatController
 {
     public function indexAction()
     {
         $searchDashboard = new SearchDashboard();
         $searchDashboard->setUser($this->Auth()->getUser());
-        $this->view->dashboard = $searchDashboard->search($this->params->get('q'));
 
-        // NOTE: This renders the dashboard twice. Remove this once we can catch exceptions thrown in view scripts.
-        $this->view->dashboard->render();
+        $this->controls->setTabs($searchDashboard->getTabs());
+        $this->content = $searchDashboard->search($this->getParam('q'));
```

Comment:
I don't want to add a content but rather replace the default content by that from search dashboard.

Reply:
Which default content?

Reply:
And even then, there is `setContent()`.

Reply:
> Which default content?

See [here](https://github.com/Icinga/ipl-web/blob/master/src/Compat/CompatController.php#L60).

> And even then, there is `setContent()`.

Couldn’t found that in `CompatController`.

Reply:
Aaah, of course `setContent` isn't there. But anyway, the controllers `$content` shouldn't be overridden. There is already an id associated with it, which gets lost now. This may break various other things.

Use `addContent`. Wrap the dashlets in another div and add there `dashboard` as class. Update `widgets.less:122` and your JS accordingly.

Reply:
Wouldn't it be simpler to just re-add the id if it's so essential and not to adjust all the js code where all the dashlets are loaded and css styles?

Reply:
And what about it's not just the id but another attribute or class? This isn't a property of the base class without any reason.

Reply:
Also, something may expect `$content` to be an instance of `Content`, which your dashboard is not.